pkg/repository/chunk.go

```diff
@@ -125,24 +129,30 @@ func (r *Repository) TextChunkTableName() string {
 // DeleteAndCreateChunks deletes all the chunks associated with
 // a certain source table and sourceUID, then batch inserts the new chunks
 // within a transaction.
-func (r *Repository) DeleteAndCreateChunks(ctx context.Context, sourceTable string, sourceUID uuid.UUID, chunks []*TextChunk, externalServiceCall func(chunkUIDs []string) (map[string]any, error)) ([]*TextChunk, error) {
+func (r *Repository) DeleteAndCreateChunks(
+	ctx context.Context,
+	fileUID uuid.UUID,
+	chunks []*TextChunk,
+	externalServiceCall func(chunkUIDs []string) (destinations map[string]string, _ error),
+) ([]*TextChunk, error) {
 	logger, _ := logx.GetZapLogger(ctx)
+
 	// Start a transaction
-	err := r.db.Transaction(func(tx *gorm.DB) error {
+	err := r.db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
 		// Delete existing chunks
-		where := fmt.Sprintf("%s = ? AND %s = ?", TextChunkColumn.SourceTable, TextChunkColumn.SourceUID)
```

Comment:
🗒️ We were cleaning up the chunk records by source UID. However, during the processing the sources change (first the converted file, source for the chunks, then the chunks, sources for the embeddings). ➡️ We need to clean up by file UID.